include/kodlab_mjbots_sdk/behavior.h

```diff
@@ -0,0 +1,184 @@
+/**
+ * @file behavior.h
+ * @author Ethan J. Musser (emusser@seas.upenn.edu)
+ * @brief Base class for implementing robot behaviors
+ * @date 7/22/22
+ * 
+ * @copyright Copyright 2022 The Trustees of the University of Pennsylvania. All
+ *            rights reserved.
+ * 
+ */
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <utility>
+#include "kodlab_mjbots_sdk/robot_base.h"
+#include "kodlab_mjbots_sdk/log.h"
+
+namespace kodlab
+{
+
+/**
+ * @brief Robot behavior object
+ * @tparam Robot[optional] derived `kodlab::RobotBase` class
+ */
+template<class Robot = kodlab::RobotBase>
+class Behavior
+{
+  static_assert(std::is_base_of<kodlab::RobotBase, Robot>::value,
+                "Robot must have base kodlab::RobotBase.");
+
+protected:
+  /**
+   * @brief Robot that this behavior runs on
+   */
+  std::shared_ptr<Robot> robot_;
+
+  /**
+   * @brief Behavior name, default is "OFF"
+   */
+  std::string name_;
+
+  /**
+   * @brief Initialization state of this behavior
+   * @note Must be updated appropriately for integration with BehaviorManager
+   */
+  bool initialized_ = false;
+
+  /**
+   * @brief Active state of this behavior
+   */
+  bool active_ = false;
+
+public:
+  /**
+   * @brief Construct a Behavior object
+   * @param robot robot behavior is executing on
+   * @param name[optional] behavior name
+   */
+  Behavior(std::shared_ptr<Robot> robot, std::string name = "EMPTY")
+      : robot_(robot), name_(std::move(name)) {}
+
+  /**
+   * @brief Destructor
+   */
+  virtual ~Behavior() {}
```

Comment:
Whats wrong with the default Destructor?

Reply:
This fixed an issue when stopping the control loop that some memory was freed and then accessed.  It didn't affect performance, but I figured it was worth resolving.

It has to do with deleting derived class instances through pointers to their base.  Check out the 'Virtual Destructors' section of [this link](https://en.cppreference.com/w/cpp/language/destructor).

Reply:
This makes the destructor virtual which is necessary when you plan to use upcasted pointers.

Reply:
Thats good to know. Can you make an issue to propagate this to other inherited classes. Might fix our segfault

Reply:
Will do!

Reply:
#49